OpenRA.Mods.D2k/Activities/PickupUnit.cs

```diff
@@ -0,0 +1,124 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2014 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+using System;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.D2k.Traits;
+using OpenRA.Mods.RA;
+using OpenRA.Mods.RA.Activities;
+using OpenRA.Mods.RA.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.D2k.Activities
+{
+	public class PickupUnit : Activity
+	{
+		readonly Actor carryable;
+		readonly IMove movement;
+		readonly Carryable c;
```

Comment:
Having a variable `carryable` as a reference to an Actor type, while having a reference to a Carryable type isn't good design IMO.

I suggest `Actor toPickup` and `Carryable carryable`.